examples/chip-tool-darwin/commands/clusters/ClusterCommandBridge.h

```diff
@@ -0,0 +1,135 @@
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#pragma once
+
+#import <CHIP/CHIP.h>
+#import <CHIP/CHIPDevice_Internal.h>
+#include <lib/support/UnitTestUtils.h>
+
+#include "ModelCommandBridge.h"
+
+class ClusterCommand : public ModelCommand {
+public:
+    ClusterCommand()
+        : ModelCommand("command-by-id")
+    {
+        AddArgument("cluster-id", 0, UINT32_MAX, &mClusterId);
+        AddArgument("command-id", 0, UINT32_MAX, &mCommandId);
+        AddArgument("payload", &mPayload);
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+        AddArgument("repeat-count", 1, UINT16_MAX, &mRepeatCount);
+        AddArgument("repeat-delay-ms", 0, UINT16_MAX, &mRepeatDelayInMs);
+        ModelCommand::AddArguments();
+    }
+
+    ClusterCommand(chip::ClusterId clusterId)
+        : ModelCommand("command-by-id")
+        , mClusterId(clusterId)
+    {
+        AddArgument("command-id", 0, UINT32_MAX, &mCommandId);
+        AddArgument("payload", &mPayload);
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+        AddArgument("repeat-count", 1, UINT16_MAX, &mRepeatCount);
+        AddArgument("repeat-delay-ms", 0, UINT16_MAX, &mRepeatDelayInMs);
+        ModelCommand::AddArguments();
+    }
+
+    ClusterCommand(const char * _Nonnull commandName)
+        : ModelCommand(commandName)
+    {
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+        AddArgument("repeat-count", 1, UINT16_MAX, &mRepeatCount);
+        AddArgument("repeat-delay-ms", 0, UINT16_MAX, &mRepeatDelayInMs);
+    }
+
+    ~ClusterCommand() {}
+
+    CHIP_ERROR SendCommand(CHIPDevice * _Nonnull device, chip::EndpointId endpointId) override
+    {
+        chip::TLV::TLVWriter writer;
+        chip::TLV::TLVReader reader;
+        NSDictionary * commandFieldDict = @{@"type" : @"Structure", @"value" : @[]};
+
+        mData = static_cast<uint8_t *>(chip::Platform::MemoryCalloc(sizeof(uint8_t), mDataMaxLen));
+        VerifyOrReturnError(mData != nullptr, CHIP_ERROR_NO_MEMORY);
+
+        writer.Init(mData, mDataMaxLen);
+
+        ReturnErrorOnFailure(mPayload.Encode(writer, chip::TLV::AnonymousTag()));
+        reader.Init(mData, writer.GetLengthWritten());
+        ReturnErrorOnFailure(reader.Next());
+
+        id commandFields = NSObjectFromCHIPTLV(&reader);
+        if (commandFields != nil) {
```

Comment:
I guess the real question here is whether failure to decode should error out instead of falling back to a default...

Reply:
Yes, that makes sense. I must have not understood. I'll update this and write to error out when a nil is returned. The only thing is the I do not have the error occurred. So I will have to return `CHIP_ERROR_INTERAL`

Reply:
Oook. Updated and tested. I couldn't seem to get the failure scenario to occur though.